### PR TITLE
Issue 748: conflit with gpnupg2 on centos8

### PR DESCRIPTION
**Pull request by haidars**
Issue #748 

- Add guards on while installing gnupg to avoid conflit with gnupg2 on RHEL/Centos8

### CODE REVIEW

**Review by mx-psi — CHANGES_REQUESTED** [review]
Thanks for the PR! The PR code looks good to me but you will need the CI pass so that we can merge this. In particular you need to update the spec tests to account for this change. You can run the tests locally following [these guidelines](https://github.com/DataDog/chef-datadog/blob/master/CONTRIBUTING.md#testing).

File: .rubocop.yml [hunk 1]
```diff
@@ -21,9 +21,6 @@ Style/TrailingCommaInArrayLiteral:
 Style/TrailingCommaInHashLiteral:
   Enabled: false
 
-Style/TrailingCommaInLiteral:
-  Enabled: false
-
```
[the remark this answers is not in the document]
Comment (haidars) [reply .rubocop.yml:26]: it was intended as this cop has been deprecated


```bash
 h.salauroo@lion-hsa2  ~/Documents/TnTC/chef/chef-datadog   gnupg ✚  rubocop -v
0.86.0
 h.salauroo@lion-hsa2  ~/Documents/TnTC/chef/chef-datadog   gnupg ✚  rubocop .
Error: The `Style/TrailingCommaInLiteral` cop has been removed. Please use `Style/TrailingCommaInArrayLiteral` and/or `Style/TrailingCommaInHashLiteral` instead.
(obsolete configuration found in .rubocop.yml, please update it)
 ✘ h.salauroo@lion-hsa2  ~/Documents/TnTC/chef/chef-datadog   gnupg ✚  grep -EA 1 'TrailingCommaInArrayLiteral|TrailingCommaInHashLiteral' .rubocop.yml
Style/TrailingCommaInArrayLiteral:
  Enabled: false
--
Style/TrailingCommaInHashLiteral:
  Enabled: false
 h.salauroo@lion-hsa2  ~/Documents/TnTC/chef/chef-datadog   gnupg ✚ 
```
going to test the CI and check again

[the remark this answers is not in the document]
Comment (albertvaka) [reply .rubocop.yml:26]: Our CI runs tests on old  versions of Chef, which need old versions of Ruby, which aren't compatible with the latest Rubocop, so we probably want to keep that cop around for a while until we stop supporting them.

**Comment by haidars** [comment]
> Thanks for the PR! The PR code looks good to me but you will need the CI pass so that we can merge this. In particular you need to update the spec tests to account for this change. You can run the tests locally following [these guidelines](https://github.com/DataDog/chef-datadog/blob/master/CONTRIBUTING.md#testing).

Spec file has been updated to reflect the changes.

File: .rubocop.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (haidars) [reply .rubocop.yml]: change to rubocop.yml has been rollback

File: spec/repository_spec.rb [hunk 2]
```diff
@@ -41,10 +41,6 @@
         )
       end
 
-      it 'installs gnupg' do
-        expect(chef_run).to install_package('gnupg')
-      end
-
```
Comment (mx-psi) [inline spec/repository_spec.rb:47]: We still install `gnupg` on some situations so we want to keep the spec test. One way to reflect what you did in the code is:

```
      it 'installs gnupg' do
        expect(chef_run).to install_package('gnupg') if chef_run.node['packages']['gnupg2'].nil?
      end
```

Would you mind adding back the spec tests following that pattern?

Comment (haidars) [reply spec/repository_spec.rb:47]: Of course... will do that in the evening...

Comment (haidars) [reply spec/repository_spec.rb]: Done for unit test

Comment (mx-psi) [reply spec/repository_spec.rb]: Awesome, thanks for taking the time to fix that!

**Review by mx-psi — APPROVED** [review]
LGTM, thank you again! 🚀

**Pull request merged**